File: app/list/print_list_reverse.py
# coding=utf-8
# author: Xiguang Liu<g10guang@gmail.com>
# 2018-04-26 17:38
# 题目描述：https://www.nowcoder.com/practice/d0267f7f55b3412ba93bd35cfa8e8035?tpId=13&tqId=11156&tPage=1&rp=1&ru=/ta/coding-interviews&qru=/ta/coding-interviews/question-ranking

# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    # 返回从尾部到头部的列表值序列，例如[1,2,3]
    def printListFromTailToHead(self, listNode):
        """
        递归输出
        :param listNode:
        :return:
        """
        if not hasattr(self, 'result'):
            self.result = []
        if listNode:
            self.printListFromTailToHead(listNode.next)
            self.result.append(listNode.val)

        return self.result

File: app/list/print_list_reverse.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

/**
 * 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
 *
 * @param head ListNode类
 * @return int整型一维数组
 */
// https://www.nowcoder.com/practice/d0267f7f55b3412ba93bd35cfa8e8035?tpId=13&tqId=11156&tPage=1&rp=1&ru=/ta/coding-interviews&qru=/ta/coding-interviews/question-ranking
func printListFromTailToHead(head *ListNode) []int {
	result := make([]int, 0, 16)
	var fn func(node *ListNode)
	fn = func(node *ListNode) {
		if node == nil {
			return
		}
		fn(node.Next)
		result = append(result, node.Val)
	}
	fn(head)
	return result
}
